### ISSUE TITLE
chore(repo): revert mergeCommitAllowed sync-repo-setting

### ISSUE THREAD

**Issue by noahdietz** [googleapis/google-cloud-go#6369]
Once the work with `storage-refactor` branch is complete and the branch is merged into `main`, we must again disable `mergeCommitAllowed` and remove the branch settings for `storage-refactor`. These changes were introduced in https://github.com/googleapis/google-cloud-go/pull/6357.
